### PR TITLE
Fix diagrams

### PR DESCRIPTION
**Pull request by JulietTeoh**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -64,14 +64,19 @@ The sections below give more details of each component.
...
+The UI consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `ResultDisplay`, `EndpointListPanel`, 
+`StatusBarFooter` etc. All these, including the `MainWindow`, inherit from the abstract `UiPart` class.
 
-The `UI` component uses JavaFx UI framework. The layout of these UI parts are defined in matching `.fxml` files that are in the `src/main/resources/view` folder. For example, the layout of the [`MainWindow`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/ui/MainWindow.java) is specified in [`MainWindow.fxml`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/resources/view/MainWindow.fxml)
+The `UI` component uses JavaFx UI framework. The layout of these UI parts are defined in matching `.fxml` files that 
+are in the `src/main/resources/view` folder. For example, the layout of the 
+[`MainWindow`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/ui/MainWindow.java) 
+is specified in 
+[`MainWindow.fxml`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/resources/view/MainWindow.fxml)
```
Comment (tjtanjin) [inline docs/DeveloperGuide.md:79]: Need to update the links here :D

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -92,29 +97,28 @@ The `UI` component,
...
-* exposes an unmodifiable `ObservableList<Person>` that can be 'observed' e.g. the UI can be bound to this list so that the UI automatically updates when the data in the list change.
+* exposes an unmodifiable `ObservableList<Endpoint>` that can be 'observed' e.g. the UI can be bound to this list so 
+  that the UI automatically updates when the data in the list change.
 * does not depend on any of the other three components.
 
-todo
-<div markdown="span" class="alert alert-info">:information_source: **Note:** An alternative (arguably, a more OOP) model is given below. It has a `Tag` list in the `AddressBook`, which `Person` references. This allows `AddressBook` to only require one `Tag` object per unique `Tag`, instead of each `Person` needing their own `Tag` object.<br>
+<div markdown="span" class="alert alert-info">:information_source: **Note:** An alternative (arguably, a more OOP) model
+is given below. It has a `Tag` list in the `AddressBook`, which `Person` references. This allows `AddressBook` to only 
+require one `Tag` object per unique `Tag`, instead of each `Person` needing their own `Tag` object.<br>
```
Comment (tjtanjin) [inline docs/DeveloperGuide.md:120]: Need to replace instances of `AddressBook` and `Person` :D

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -139,47 +143,39 @@ Classes used by multiple components are in the `seedu.us.among.commons` package.
...
 * is stored in `EndpointList` of the `Model`
 * encapsulates an API endpoint
 
-An `Endpoint` contains the following components:
-1. a `Method` which represents the type of request an API endpoint will send to the server
-2. a `Address` which represents the address to which the API request is made
-3. a `Data` which represents the data that is to be sent to the server when an API request is made.
+An `Endpoint` contains the following attributes:
+1. a `Method`, which represents the type of request an API endpoint will send to the server
+2. a `Address`, which represents the address to which the API request is made
```
Comment (tjtanjin) [inline docs/DeveloperGuide.md:155]: Minor thing but "an `Address`" might sound better :3

**Review by tjtanjin — APPROVED** [review]
LGTM, just some nits to fix! :)

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (JulietTeoh) [reply docs/DeveloperGuide.md:79]: I will resolve in another pr :>

**Pull request merged**